src/main/static/src/components/henkilo/OppijaViewContainer.js

```diff
@@ -18,16 +18,26 @@ class OppijaViewContainer extends React.Component {
             this.props.router.replace(`/admin/${this.props.oidHenkilo}`);
         }
         else {
-            this.props.updateHenkiloNavigation(oppijaNavi(this.props.oidHenkilo));
+            this.fetch(this.props.oidHenkilo)
+        }
+    }
 
-            await this.props.fetchHenkilo(this.props.oidHenkilo);
-            this.props.fetchHenkiloSlaves(this.props.oidHenkilo);
-            this.props.fetchYhteystietotyypitKoodisto();
-            this.props.fetchKieliKoodisto();
-            this.props.fetchKansalaisuusKoodisto();
+    async componentWillReceiveProps(nextProps) {
+        if (nextProps.oidHenkilo !== this.props.oidHenkilo) {
+            this.fetch(nextProps.oidHenkilo)
         }
     }
 
+    async fetch(oid) {
+        this.props.updateHenkiloNavigation(oppijaNavi(oid));
+
+        await this.props.fetchHenkilo(oid);
```

Comment:
Onko noiden neljän kutsun välttämättä odotettava henkilön tietojen hakemista? Jos on niin tämä on hyvä näin, mutta näyttäisi, että tässä ei ole tarvetta käyttää await:a ollenkaan.